test/Orleans.SyncWork.Tests/SerializationSurrogates/SaltParseExceptionSerializationSurrogate.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using BCrypt.Net;
+
+namespace Orleans.SyncWork.Tests.SerializationSurrogates;
+
+[GenerateSerializer]
+public struct SaltParseExceptionSerializationSurrogate
```

Comment:
How does this get used/registered?  Is it just auto magic?

Reply:
Yes -- Orleans v7 did away with the explicit "AddApplicationParts()" construct and does assembly scanning automatically. These serializer attributes trigger the Orleans code to generate serializers for these classes. You will see runtime errors if you attempt to pass a class between grains and you've missed the `[GenerateSerializer]` attribute, something along the lines of "could not locate serializer for type".

https://learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans#packaging-changes
https://learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans#serialization